src/wolframite/base/parse.clj

```diff
@@ -119,10 +119,11 @@
 ;; parameters list used to be: [expr & [type]] (??)
 (defn parse-simple-vector [expr type {:keys [flags] :as opts}]
   (let [type (or type (simple-vector-type expr))]
-    (if (and (options/flag?' flags :N)
+    (if (and (options/flag?' flags :arrays)
+             ;; TODO Why only these types? W. supports boolean, byte, char, short, int, long, float, double, String arrays;
+             ;; Though only byte, short, int, float, double have "fast" methods
              (some #{:Expr/INTEGER :Expr/BIGINTEGER :Expr/REAL :Expr/BIGDECIMAL} #{type}))
-      ((if (options/flag?' flags :vectors) vec seq)

```

Comment:
dropping this line is the essential change - with it, we always turned the array into a vector or a seq, thus turning it into a less efficient structure.